terraform/gcp-constellation/main.tf

```
@@ -0,0 +1,86 @@
+locals {
+  region = substr(var.zone, 0, length(var.zone) - 2)
+}
+
+module "gcp_iam" {
+  source             = "../infrastructure/iam/gcp"
+  project_id         = var.project
+  service_account_id = var.service_account_id
+  region             = local.region
+  zone               = var.zone
+}
+
+
+resource "null_resource" "ensure_yq" {
+  provisioner "local-exec" {
+    command = <<EOT
+         ../constellation-cluster/install-yq.sh
+    EOT
+  }
+  triggers = {
+    always_run = timestamp()
+  }
+}
+
+module "fetch_image" {
+  source              = "../fetch-image"
+  csp                 = "gcp"
+  attestation_variant = "gcp-sev-es"
+  image               = var.image
+  depends_on          = [null_resource.ensure_yq]
+}
+
+
+module "gcp" {
+  source          = "../infrastructure/gcp"
+  project         = var.project
+  image_id        = module.fetch_image.image
+  name            = var.name
+  node_groups     = var.node_groups
+  region          = local.region
+  zone            = var.zone
+  debug           = var.debug
+  custom_endpoint = var.custom_endpoint
+}
+
+resource "null_resource" "sa_account_file" {
```

```suggestion
resource "null_resource" "service_account_file" {
```
sa_account is semantically duplicate (i.e. serviceaccount_account)